solr/core/src/java/org/apache/solr/cloud/api/collections/Assign.java

```diff
@@ -180,7 +180,7 @@ private static int defaultCounterValue(DocCollection collection, boolean newColl
     if (newCollection) return 0;
 
     int defaultValue;
-    if (collection.getSlice(shard) != null && collection.getSlice(shard).getReplicas().isEmpty()) {
+    if (collection == null || collection.getSlice(shard) != null && collection.getSlice(shard).getReplicas().isEmpty()) {
```

Comment:
```suggestion
    if (collection == null || (collection.getSlice(shard) != null && collection.getSlice(shard).getReplicas().isEmpty())) {
```

or doing `if (newCollection || collection == null) return 0;` at line 180 above might be slightly easier to comprehend.

Reply:
👍

Reply:
```suggestion
    if (collection.getSlice(shard) != null && collection.getSlice(shard).getReplicas().isEmpty()) {
```

now that line 180 has the null check.